process/coordinator/process.go

```diff
@@ -1480,10 +1519,17 @@ func (tc *transactionCoordinator) verifyFees(
 		totalMaxDeveloperFees.Add(totalMaxDeveloperFees, scheduledGasAndFees.DeveloperFees)
 	}
 
-	for _, miniBlock := range body.MiniBlocks {
+	for index, miniBlock := range body.MiniBlocks {
 		if miniBlock.Type == block.PeerBlock {
 			continue
 		}
+		if tc.flagScheduledMiniBlocks.IsSet() {
+			miniBlockHeader := header.GetMiniBlockHeaderHandlers()[index]
```

Comment:
here as well, maybe add a check for index out of bounds.

Reply:
Done